core/src/main/scala/pickling/package.scala

```diff
@@ -5,14 +5,67 @@ import scala.language.experimental.macros
 
 package object pickling {
 
+  def unpickle[T](thePickle: Pickle)(implicit unpickler: Unpickler[T], format: PickleFormat): T = {
+  	// TODO - move GRL locking code into just the runtime unpickler code + generators.
+  	internal.GRL.lock()
+  	try {
+  	  val reader = format.createReader(thePickle.asInstanceOf[format.PickleType])
+  	  val result = unpickler.unpickleEntry(reader).asInstanceOf[T]
+  	  // TODO - some mechanism to disable this.
+  	  internal.clearUnpicklees();
+  	  result
+  	} finally internal.GRL.unlock()
+  }
+  def pickle[T](picklee: T)(implicit format: PickleFormat, pickler: SPickler[T]): format.PickleType = {
+  	val builder = format.createBuilder
+  	pickleInto(picklee, builder)
+  	// TODO - some mechanism to disable this.
+  	internal.clearPicklees()
+  	builder.result.asInstanceOf[format.PickleType]
+  }
+  def pickleInto[T](picklee: T, builder: PBuilder)(implicit pickler: SPickler[T]): Unit = {
+  	// TODO - BeginEntry/EndEntry needed?
+  	// TODO - this hinting should be in the pickler, not here.  We need to understand
+  	//        when we want to use this vs. something else.
+  	// TODO - Move GRL Lock only into dynamic picklers
+  	internal.GRL.lock()
+  	try { 
+  	  if(null == picklee) {
+  	    builder.hintTag(FastTypeTag.Null)
+  	    all.nullPicklerUnpickler.pickle(null, builder)
+  	  } else {
+  	    builder.hintTag(pickler.tag)
+  	    pickler.pickle(picklee, builder)
+  	  }
+  	  } finally internal.GRL.unlock()
+    // TODO - This usually doesn't clear Picklee's, but should we?
```

Comment:
Correct. It should not clear picklees, since `pickleInto` is supposed to be used only for pickling nested objects, never top-level objects. Picklees should be cleared only after pickling a top-level object. Thus, it's correct not to clear picklees at this point.